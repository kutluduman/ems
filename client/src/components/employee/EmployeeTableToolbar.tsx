import { styled } from "@mui/material/styles";
import Box from "@mui/material/Box";
import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
import DialogContentText from "@mui/material/DialogContentText";
import IconButton from "@mui/material/IconButton";
import InputAdornment from "@mui/material/InputAdornment";
import OutlinedInput from "@mui/material/OutlinedInput";
import SearchIcon from "@mui/icons-material/Search";
import Toolbar from "@mui/material/Toolbar";
import Tooltip from "@mui/material/Tooltip";
import Typography from "@mui/material/Typography";

import { useState } from "react";
import Modal, { ModalProps } from "../Modal";

const RootStyle = styled(Toolbar)(({ theme }) => ({
  height: 96,
  display: "flex",
  justifyContent: "space-between",
  padding: theme.spacing(0, 1, 0, 3),
}));

const SearchStyle = styled(OutlinedInput)(
  ({ theme }: { [key: string]: any }) => ({
    width: 240,
    transition: theme.transitions.create(["box-shadow", "width"], {
      easing: theme.transitions.easing.easeInOut,
      duration: theme.transitions.duration.shorter,
    }),
    "&.Mui-focused": { width: 320, boxShadow: theme.customShadows?.z8 },
    "& fieldset": {
      borderWidth: `1px !important`,
      borderColor: `${theme.palette.grey[500_32]} !important`,
    },
  })
);

interface EmployeeTableToolbarProps {
  numSelected: number;
  filterName: string;
  onFilterName: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void;
  handleMultipleUsersDelete: Function;
}

const EmployeeTableToolbar = ({
  numSelected,
  filterName,
  onFilterName,
  handleMultipleUsersDelete,
}: EmployeeTableToolbarProps) => {
  const [open, setOpen] = useState(false);

  return (
    <RootStyle
      sx={{
        ...(numSelected > 0 && {
          color: "primary.main",
          bgcolor: "primary.lighter",
        }),
      }}
    >
      {numSelected > 0 ? (
        <Typography component="div" variant="subtitle1">
          {numSelected} selected
        </Typography>
      ) : (
        <SearchStyle
          value={filterName}
          onChange={onFilterName}
          placeholder="Search employee..."
          startAdornment={
            <InputAdornment position="start">
              <Box component={SearchIcon} sx={{ color: "text.disabled" }} />
            </InputAdornment>
          }
        />
      )}

      {numSelected > 0 ? (
        <Tooltip title="Delete" onClick={() => setOpen(true)}>
          <IconButton>
            <DeleteOutlineIcon />
          </IconButton>
        </Tooltip>
      ) : null}

      <Modal
        title="Are you sure?"
        actions={[
          {
            title: "No",
            action: () => setOpen(false),
          },
          {
            title: "Yes",
            action: () => {
              setOpen(false);
              handleMultipleUsersDelete();
            },
          },
        ]}
        open={open}
        setOpen={setOpen as never as ModalProps["setOpen"]}
      >
        <DialogContentText id="alert-dialog-description">
          Are you sure you want to delete {numSelected} Employees?
        </DialogContentText>
      </Modal>
    </RootStyle>
  );
}
export default EmployeeTableToolbar;