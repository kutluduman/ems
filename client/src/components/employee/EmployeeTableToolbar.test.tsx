import { create } from "react-test-renderer";

import EmployeeTableToolbar from "./EmployeeTableToolbar";

describe("EmployeeTableToolbar Component", () => {
  test("renders the EmployeeTableToolbar Component", () => {
    const tree = create(
      <EmployeeTableToolbar
        numSelected={10}
        filterName={"filterName"}
        handleMultipleUsersDelete={() => {}}
        onFilterName={(
          event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
        ) => {}}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});