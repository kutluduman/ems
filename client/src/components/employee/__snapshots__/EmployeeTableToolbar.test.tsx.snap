// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeTableToolbar Component renders the EmployeeTableToolbar Component 1`] = `
<div
  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-pux2q4-MuiToolbar-root"
>
  <div
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    10
     selected
  </div>
  <button
    aria-label="Delete"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="DeleteOutlineIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
